diff --git a/src/animation/tracks/QuaternionKeyframeTrack.d.ts b/src/animation/tracks/QuaternionKeyframeTrack.d.ts
index 4bcaa98982ac38644505fa583d2d9b0f4638ad94..f1b9eac7c0fe6c23785e0e073cdf9e8d02b01681 100644
--- a/src/animation/tracks/QuaternionKeyframeTrack.d.ts
+++ b/src/animation/tracks/QuaternionKeyframeTrack.d.ts
@@ -9,8 +9,8 @@ export class QuaternionKeyframeTrack extends KeyframeTrack {
      * Constructs a new Quaternion keyframe track.
      *
      * @param {string} name - The keyframe track's name.
-     * @param {Array<number>} times - A list of keyframe times.
-     * @param {Array<number>} values - A list of keyframe values.
+     * @param {ArrayLike<number>} times - A list of keyframe times.
+     * @param {ArrayLike<number>} values - A list of keyframe values.
      * @param {(InterpolateLinear|InterpolateDiscrete|InterpolateSmooth)} [interpolation] - The interpolation type.
      */
     constructor(name: string, times: ArrayLike<number>, values: ArrayLike<number>, interpolation?: InterpolationModes);
diff --git a/src/nodes/core/LightingModel.d.ts b/src/nodes/core/LightingModel.d.ts
index cb894c04eee863c51de94dc33e0e6ca2aaffcd82..fbbc46fefdb6c74a0c12fc0991b67f4fe6050d00 100644
--- a/src/nodes/core/LightingModel.d.ts
+++ b/src/nodes/core/LightingModel.d.ts
@@ -10,6 +10,8 @@ export interface LightingModelReflectedLight {
 }
 
 export interface LightingModelDirectInput extends DirectLightData {
+    lightColor: Node;
+    lightDirection: Node;
     lightNode: Node;
     reflectedLight: LightingModelReflectedLight;
 }
diff --git a/src/nodes/core/NodeBuilder.d.ts b/src/nodes/core/NodeBuilder.d.ts
index eb477e39e05c5e9e909caa7b70f6474209627fe2..0be5b27dcad2d46fcc161ed105400b805c11c16c 100644
--- a/src/nodes/core/NodeBuilder.d.ts
+++ b/src/nodes/core/NodeBuilder.d.ts
@@ -49,6 +49,8 @@ export default abstract class NodeBuilder {
     cache: NodeCache;
     globalCache: NodeCache;
 
+    context: NodeBuilderContext;
+
     /**
      * @TODO used to be missing. check the actual type later
      */
diff --git a/src/nodes/tsl/TSLCore.d.ts b/src/nodes/tsl/TSLCore.d.ts
index 0e3c220817408f06f605a06821afd952273e38bc..30f3901e32339e14a68870abb9a69d3b2aaf2bd2 100644
--- a/src/nodes/tsl/TSLCore.d.ts
+++ b/src/nodes/tsl/TSLCore.d.ts
@@ -391,6 +391,7 @@ export const bvec4: (node: Node) => ShaderNodeObject<Node>;
 interface Matrix2Function {
     (value: Matrix2): ShaderNodeObject<ConstNode<Matrix2>>;
     (node: Node): ShaderNodeObject<Node>;
+    (n11: Node | number, n12: Node | number, n21?: Node | number, n22?: Node | number): ShaderNodeObject<ConstNode<Matrix2>>;
 }
 
 export const mat2: Matrix2Function;
@@ -443,7 +444,7 @@ export const mat4: Matrix4Function;
 export const string: (value?: string) => ShaderNodeObject<ConstNode<string>>;
 export const arrayBuffer: (value: ArrayBuffer) => ShaderNodeObject<ConstNode<ArrayBuffer>>;
 
-export const element: (node: Node, indexNode: Node) => ShaderNodeObject<Node>;
+export const element: (node: Node | number, indexNode: Node | number) => ShaderNodeObject<Node>;
 export const convert: (node: Node, types: string) => ShaderNodeObject<Node>;
 export const split: (node: Node, channels?: string) => ShaderNodeObject<Node>;
 
diff --git a/src/textures/ExternalTexture.d.ts b/src/textures/ExternalTexture.d.ts
index 8f62cacbb1cab570373a0f4d098f967a675bd4b2..0e0d8077380bbe4f838b7e97e487925ca1521cd5 100644
--- a/src/textures/ExternalTexture.d.ts
+++ b/src/textures/ExternalTexture.d.ts
@@ -1,3 +1,5 @@
+/// <reference types="@webgpu/types" />
+
 import { Texture } from "./Texture.js";
 
 declare class ExternalTexture extends Texture {
